### DISCUSSION TITLE
Gumroad automatic download

### DISCUSSION THREAD

**Discussion by Sim0nW0lf** [kimai/Paid-plugins#63] [General]
Hi,

is there a way to download the plugins automatically from Gumroad?
I have done it with your GitHub repositories before but since they're not available anymore I would like to integrate that with Gumroad.
I have not found any API and have no idea how I could try doing that unfortunately.

best regards,
Simon

**Comment by kevinpapst** [comment]
Why would you download the plugins automatically ? Once after an updated was published sounds enough.
I don't see the use-case, could you explain what you are trying to achieve?

**Comment by Sim0nW0lf** [comment]
Once after an upload is published is fine but I manage lots of containers and to make it easier I just automate all upgrade processes.

At the moment every time a plugin is updated I have to download it, transfer it to the server and launch my update script.
When it was hosted on github for example I didn't do anything. It just downloaded all the new files and updated by itself in the night.

**Reply by kevinpapst** [reply]
That process sounds scary too me. Unfortunately I don't know of any Gumraod API that could be used. 
And even though I understand your use-case, I do not think this is something that I can solve.

**Reply by Sim0nW0lf** [reply]
Okay, np. I just wanted to know if I was missing something when researching.

**Comment by rbuehler-teletrend** [comment]
I had been using git clone too, to get the latest plugin versions directly (although not automated).
It was easier than manually download and upload to the plugin directory.

**Comment by kevinpapst** [comment]
I hear you! I understand that such a change is not welcome. And I am sorry if it causes trouble!

The previous workflow was okay in the early beginning, but with a growing user base, processes need to be revised and changed if they don't scale. Managing several repositories manually was just eating up too much of my time.

I really want to spent that time on building and shipping a great next Kimai version for everyone.

**Comment by rbuehler-teletrend** [comment]
I just wanted to point out a use-case. For me it's not a big issue and it will be fine to download from Gumroad.
